consensus/validation_fuzz.go

```diff
@@ -3,21 +3,54 @@
 package consensus
 
 import (
-	"bytes"
+	"crypto/ed25519"
 	"go.sia.tech/core/types"
-	"io"
 )
 
 func Fuzz(data []byte) int {
-	d := types.NewDecoder(io.LimitedReader{bytes.NewReader(data), 1e7})
+	d := types.NewBufDecoder(data)
 
 	var txn types.Transaction
 	txn.DecodeFrom(d)
 	if err := d.Err(); err != nil {
 		return -1
 	}
 
+	var merkleTxn types.Transaction
+	for _, input := range txn.SiacoinInputs {
+		merkleTxn.SiacoinOutputs = append(merkleTxn.SiacoinOutputs, types.SiacoinOutput{
+			Value:   input.Parent.Value,
+			Address: input.Parent.Address,
+		})
+	}
+	for _, input := range txn.SiafundInputs {
+		merkleTxn.SiafundOutputs = append(merkleTxn.SiafundOutputs, types.SiafundOutput{
+			Value:   input.Parent.Value,
+			Address: input.Parent.Address,
+		})
+	}
```

Comment:
good start. You'll also need to do this for file contracts.

The `Element` types embed their corresponding `Output` types, so you should be able to write:
```go
	for i, sci := range txn.SiacoinInputs {
		setupTxn.SiacoinOutputs = append(setupTxn.SiacoinOutputs,  sci.Parent.SiacoinOutput)
	}
	for i, sfi := range txn.SiafundInputs {
		setupTxn.SiafundOutputs = append(setupTxn.SiafundOutputs,  sfi.Parent.SiafundOutput)
	}
	for i, fcr := range txn.FileContractRevisions {
		setupTxn.FileContracts = append(setupTxn.FileContracts,  fcr.Parent.FileContract)
	}
	for i, fcr := range txn.FileContractResolutions {
		setupTxn.FileContracts = append(setupTxn.FileContracts,  fcr.Parent.FileContract)
	}
```

You'll also need to sign the contract revisions, and the resolutions will need valid proofs. So, on second thought, maybe for now just set `FileContractResolutions` to `nil` and we'll deal with them later. 😅